File: src/main/java/model/UserCouponUsage.java
package model;

import jakarta.persistence.*;
import org.hibernate.annotations.ColumnDefault;

import java.time.Instant;
import java.util.UUID;

@Entity
public class UserCouponUsage {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "UsageID", nullable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "UserID", nullable = false)
    private User userID;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "CouponID", nullable = false)
    private Coupon couponID;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "OrderID", nullable = false)
    private Order orderID;

    @ColumnDefault("getdate()")
    @Column(name = "UsedDate")
    private Instant usedDate;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public User getUserID() {
        return userID;
    }

    public void setUserID(User userID) {
        this.userID = userID;
    }

    public Coupon getCouponID() {
        return couponID;
    }

    public void setCouponID(Coupon couponID) {
        this.couponID = couponID;
    }

    public Order getOrderID() {
        return orderID;
    }

    public void setOrderID(Order orderID) {
        this.orderID = orderID;
    }

    public Instant getUsedDate() {
        return usedDate;
    }

    public void setUsedDate(Instant usedDate) {
        this.usedDate = usedDate;
    }

}

File: src/main/java/model/UserCouponUsage.kt
package model

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.util.*

@Entity
open class UserCouponUsage {
    @Id
    @ColumnDefault("newid()")
    @Column(name = "UsageID", nullable = false)
    open var id: UUID? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "UserID", nullable = false)
    open var userID: User? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "CouponID", nullable = false)
    open var couponID: Coupon? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "OrderID", nullable = false)
    open var orderID: Order? = null

    @ColumnDefault("getdate()")
    @Column(name = "UsedDate")
    open var usedDate: Instant? = null
}